exercise-001-corpus/word_count.go

```diff
@@ -0,0 +1,76 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"sort"
+	"strings"
+)
+
+type Pair struct {
+	Key   string
+	Value int
+}
+
+type PairList []Pair
+
+func (p PairList) Len() int           { return len(p) }
+func (p PairList) Swap(i, j int)      { p[i], p[j] = p[j], p[i] }
+func (p PairList) Less(i, j int) bool { return p[i].Value > p[j].Value }
+
+func sortMapByValue(m map[string]int) PairList {
+	p := make(PairList, len(m))
+	i := 0
+	for k, v := range m {
+		p[i] = Pair{k, v}
+		i++
+	}
+	sort.Sort(p)
+	return p
+}
+
+func output(p PairList) {
+	for _, v := range p {
+		fmt.Printf("%d %s \n", v.Value, v.Key)
+	}
+}
+
+func normalize(str, chr string) string {
+	return strings.Map(func(r rune) rune {
+		if strings.IndexRune(chr, r) < 0 {
+			return r
+		}
+		return -1
+	}, str)
+}
+
+func main() {
+	if len(os.Args) < 2 {
+		fmt.Println("Missing parameter, provide file name")
+	}
+
+	file, err := os.Open(os.Args[1])
+	if err != nil {
+		fmt.Print("Can't read file: ", os.Args[1])
+		panic(err)
+	}
+
+	scanner := bufio.NewScanner(file)
+	scanner.Split(bufio.ScanWords)
+
+	m := make(map[string]int)
+
+	for scanner.Scan() {
+		word := scanner.Text()
+		word = strings.ToLower(word)
+		word = normalize(word, "\".,!?")
+
+		if val, ok := m[word]; ok {
+			m[word] = val + 1
+		} else {
+			m[word] = 1
+		}
```

Comment:
Unlike in ruby, you don't have to worry that your value will be uninitialized. The default value of an `int` is 0, so you can just do `m[word] += 1`

https://play.golang.org/p/tqCkMGbjN70